### ISSUE TITLE
crashpad_handler running at 90% CPU on Mac OS

### ISSUE THREAD

**Issue by stfnhrrs** [Eloston/ungoogled-chromium#315]
After installing version 62 crashpad_handler is using up to 56% cpu on two cores, causing a system wide slowdown and my fans sound like a jumbo jet taking off.  Let me know what I can do to help debug.

**Comment by tectiv3** [comment]
yep, I have it too. Killing it won't get it started again.

**Comment by stolendata** [comment]
Can confirm same problem when running on 10.11.6. Also noticed that the mds/mdworker metadata daemons for Spotlight were running haywire during this - probably indicating high disk output by the crashpad_handler processes - though I couldn't locate exactly what or where.

**Comment by Eloston** [comment]
63 is ready for testing. If this is still a problem, please report here.

**Comment by tectiv3** [comment]
Still there...

**Comment by Eloston** [comment]
Could you guys report what arguments are passed into `crashpad_handler`? What's the parent process? Is there an instance of `run_with_crashpad` anywhere?

**Comment by Eloston** [comment]
For reference, here's documentation for Crashpad: https://chromium.googlesource.com/crashpad/crashpad

**Comment by tectiv3** [comment]
```
user           77256  50.4  0.0  2437952   6168   ??  R     1:09AM   0:16.72 /Applications/Chromium.app/Contents/Versions/63.0.3239.132/Chromium Framework.framework/Helpers/crashpad_handler --no-periodic-tasks --monitor-self-annotation=ptype=crashpad-handler --database=/Users/user/Library/Application Support/Chromium/Crashpad --annotation=plat=OS X --annotation=prod=Chromium_Mac --annotation=ver=63.0.3239.132 --handshake-fd=4
user           77254  50.3  0.0  2457936   6308   ??  S     1:09AM   0:16.68 /Applications/Chromium.app/Contents/Versions/63.0.3239.132/Chromium Framework.framework/Helpers/crashpad_handler --monitor-self --monitor-self-annotation=ptype=crashpad-handler --database=/Users/user/Library/Application Support/Chromium/Crashpad --metrics-dir=/Users/user/Library/Application Support/Chromium --annotation=plat=OS X --annotation=prod=Chromium_Mac --annotation=ver=63.0.3239.132 --handshake-fd=9
```
Quitting those two processes won't bring them back (until chromium restart)
no `run_with_crashpad` spotted.

**Comment by Eloston** [comment]
@tectiv3 Do those processes have any parents or children that should be noted?

**Comment by tectiv3** [comment]
No children, parent is `launchd`, process group - Chromium.
Actually I was wrong, sending them `SIGTERM` will quit that stuck process and re-spawn a normal one:
```
user           77256   0.0  0.0  2438976   5384   ??  S     1:09AM   0:24.01 /Applications/Chromium.app/Contents/Versions/63.0.3239.132/Chromium Framework.framework/Helpers/crashpad_handler --no-periodic-tasks --monitor-self-annotation=ptype=crashpad-handler --database=/Users/user/Library/Application Support/Chromium/Crashpad --annotation=plat=OS X --annotation=prod=Chromium_Mac --annotation=ver=63.0.3239.132 --handshake-fd=4
user           77254   0.0  0.0  2457936   5496   ??  S     1:09AM   0:27.29 /Applications/Chromium.app/Contents/Versions/63.0.3239.132/Chromium Framework.framework/Helpers/crashpad_handler --monitor-self --monitor-self-annotation=ptype=crashpad-handler --database=/Users/user/Library/Application Support/Chromium/Crashpad --metrics-dir=/Users/user/Library/Application Support/Chromium --annotation=plat=OS X --annotation=prod=Chromium_Mac --annotation=ver=63.0.3239.132 --handshake-fd=9
```

**Comment by tectiv3** [comment]
Even better, sending them `SIGTERM` will just stop them using all CPU without restarting a process :/
Notice the same PID.
BTW after that quitting Chromium won't quit those two crashpad_handler processes. They have to be killed with SIGKILL.

**Comment by Eloston** [comment]
Hmm, well it seems none of the debugging options will be easy. Either we look through the code and patches, do a debug build/make modifications and debug at runtime, or both. I haven't worked with this area of Chromium before so I have no ideas.

**Comment by tectiv3** [comment]
Can we just disable it all-together?

**Comment by Eloston** [comment]
I was thinking about it, but I'm not sure how integrated the crashpad client (or the whole client module) is in Chromium. It may not be that straight-forward to remove. That can cause some more unintended side-effects.

It's an option we can take if we aren't able to fix crashpad_handler.

**Comment by Eloston** [comment]
Could you guys check to see if dbf4c0e58765b77b9e3569facd3eee589084b0ee changes anything? I doubt this is the cause, but I'd like to be certain.

**Comment by tectiv3** [comment]
OK, I will try it a bit later.

**Comment by kelvinq** [comment]
This happens on my machine too.

**Comment by stfnhrrs** [comment]
@Eloston any progress on this bug?

**Comment by stolendata** [comment]
I have nothing to add as `dbf4c0e` unfortunately refuses to build on my El Capital setup. I'm too swamped with work right now to get enough time to dig into it. @tectiv3, have perhaps you gotten around to building it?

**Comment by Eloston** [comment]
@stfnhrrs I just finished up major rework to the build system. I'm starting development on 64, so we can test again once I merge everything in.

**Comment by tectiv3** [comment]
@stolendata nope, it refuses to build for me too.

**Comment by Artur96** [comment]
I've managed to write a patch that prevents the crashpad handler from starting. I'll create a PR once #334 is merged, also giving me time to find out whether it's stable.

**Comment by sebvargo** [comment]
I experienced this problem as well. I left my computer overnight to upload videos to a cloud using Google Chrome and my fans started to sound very loud to the point that they woke me up.

After checking Activity Monitor,  crashpad_handler was taking 199% CPU. Please see image below.

![image](https://user-images.githubusercontent.com/36892176/53565082-75a42b00-3b26-11e9-83b3-d2417c0276bd.png)

How can this be troubleshooted?

Thank you

**Comment by arutr** [comment]
@sebvargo That process shouldn't be active. Maybe the patch is no longer valid

**Comment by Eloston** [comment]
@sebvargo If you experience this with 72.0.3626.109-1, please create a new issue. Thanks.

**Comment by KimPG** [comment]
> yep, I have it too. Killing it won't get it started again.

thanks a lot.I thought my MacBook would bomb with the problem.

**Comment by simmerlee** [comment]
apple is shit! mac is shit! crap!

**Comment by krishnadamarla** [comment]
any solution ? crashpad_han running at 192 % cpu for me as well

**Comment by kramred** [comment]
@krishnadamarla, are you using the current version? Try updating first if you're still on an older version.

If the issue persists try a new profile: 
∙ make sure (ungoogled) Chromium is not running
∙ make a backup of your profile folder (should be `~/Library/Application Support/Chromium/Default` ) and then delete it, or simply rename it
∙ restart (ungoogled) Chromium

If this resolved it start migrating your settings/bookmarks etc. (always with Chromium not running)

If it still persists, please open a new issue in the macOS repo.

Remember to always have a backup first...

**Comment by JaZZim** [comment]
End the process with the command:
`kill -KILL {PID}`

**Comment by aybarsyildiz** [comment]
after ending the kernel_task from activity monitor, it fixed this situation for me

**Comment by protoboolean** [comment]
This happened to me today, macOS Big Sur (11.1). The open files of this process suggest Visual Studio Code uses it.

```
/private/var/folders/gc/📁/T/AppTranslocation/📁/d/Visual Studio Code.app/Contents/Frameworks/Electron Framework.framework/Versions/A/Helpers/chrome_crashpad_handler
/Users/am/Library/Application Support/Code/CrashpadMetrics-active.pma
/Users/am/Library/Application Support/Code/CrashpadMetrics-active.pma
```

I decided to reboot and, so far, it's behaving. Note: I don't have Chromium or Chrome installed. HTH

**Comment by exgs** [comment]
This happened to me too. mac Big Sur (11.1). Fanning is running crazy. So Force quit "Chrome crashpad_handler" in Activity Monitor

**Comment by Eloston** [comment]
I've been hearing a number of applications breaking under Big Sur, so it could be a Chromium issue. Please open an issue on ungoogled-chromium-macos if this is a problem for you instead of posting on this old issue.

**Comment by skypiea0** [comment]
> This happened to me today, macOS Big Sur (11.1). The open files of this process suggest Visual Studio Code uses it.
> 
> ```
> /private/var/folders/gc/📁/T/AppTranslocation/📁/d/Visual Studio Code.app/Contents/Frameworks/Electron Framework.framework/Versions/A/Helpers/chrome_crashpad_handler
> /Users/🏠/Library/Application Support/Code/CrashpadMetrics-active.pma
> /Users/🏠/Library/Application Support/Code/CrashpadMetrics-active.pma
> ```
> 
> I decided to reboot and, so far, it's behaving. Note: I don't have Chromium or Chrome installed. HTH

Same here on Big Sur 11.2 M1 with vs code. I have Chrome installed but it was not open at that moment. I suspect vs code too.

**Comment by stephskyy** [comment]
> > This happened to me today, macOS Big Sur (11.1). The open files of this process suggest Visual Studio Code uses it.
> > ```
> > /private/var/folders/gc/📁/T/AppTranslocation/📁/d/Visual Studio Code.app/Contents/Frameworks/Electron Framework.framework/Versions/A/Helpers/chrome_crashpad_handler
> > /Users/🏠/Library/Application Support/Code/CrashpadMetrics-active.pma
> > /Users/🏠/Library/Application Support/Code/CrashpadMetrics-active.pma
> > ```
> > 
> > 
> > I decided to reboot and, so far, it's behaving. Note: I don't have Chromium or Chrome installed. HTH
> 
> Same here on Big Sur 11.2 M1 with vs code. I have Chrome installed but it was not open at that moment. I suspect vs code too.

confirm its VScode.
Started to get this issue after installing VScode.
Uninstalling VScode got all safe and sound to normal.

**Comment by bvawdreypattern** [comment]
Had the same issue happen to me. I do not have VScode installed.

Force Quit the process in Activity Monitor fixed the issue temporarily. Every couple days it comes back and I have to force quit the process again.

**Comment by jamshid** [comment]
If you search for the process name chrome_crashpad_handler there are hits like this claiming its a virus. Hopefully that's just google spam?
https://www.cleanpcinfections.com/2021/01/how-to-remove-crashpad_handler-mac-virus-from-mac-os-jan-2021/

**Comment by knightvv** [comment]
This is 3/2023, I just hit the same issue with M2 macbook air. I installed VS Code 2 days ago, and found the process was sucking CPU and drove the CPU temperature to 180 F+. Killed it with kill -9 and now laptop is cool.

**Comment by elrinor** [comment]
Same issue on M1 Max with the latest vs code, kill -9 helps.

**Comment by mg** [comment]
Ive just experienced this on a Mac m2. The culprit was the crashpad_handler started by the Android Emulator. The instance was running at 98% cpu, with kernel at 50% cpu. Killing that specific instance (other crashpad instances are still running) brought battery drainage and head down.

The instance was running with the following flags:
`81865 ??        39:47.86 /Users/mg/Library/Android/sdk/emulator/crashpad_handler --no-rate-limit --database=/tmp/android-mg/emu-crash.db --url=https://clients2.google.com/cr/report --annotation=prod=AndroidEmulator --annotation=ver=32.1.15-10696886 --handshake-fd=67`

Android Emulator has always been problematic for me; I can leave the iOS simulator running for months but the Android Emulator needs to be shut down after a few days because it starts to drain the system. Now I am wondering if it is crashpad that is the problem.

**Comment by Leo7654** [comment]
re-spawn always, I renamed that process
```
mv /Users/hojinlee/Library/Android/sdk/emulator/crashpad_handler /Users/hojinlee/Library/Android/sdk/emulator/crashpad_handler_bk
```

**Comment by cubuspl42** [comment]
@Leo7654 I'm not sure this issue is talking about the same "instance" of `crashpad_handler`. I would guess that the Android SDK uses the same Google-created open-source program as Chromium / Chrome / Ungoogled Chrome.

Still, in my case, I also had the CPU usage issue with the Android SDK-provided `crashpad_handler` and your solution with deleting (+ backing-up) the executable binary seems to solve the problem.

**Comment by imunue1206** [comment]
it happens on my mac book air m2, is that vscode problem?

**Comment by imunue1206** [comment]
Should I give up on VSCode and switch to WebStorm?

**Comment by PF4Public** [comment]
> is that vscode problem

How is vscode related here?

**Comment by cubuspl42** [comment]
> How is vscode related here?

My wild guess is that someone observed this issue on VS Code, which is not impossible, as VS Code is based on Electron, which is based on CEF, which is based on Chromium
